test/unit/TPML-marshal.c

```diff
@@ -43,20 +42,18 @@ tpml_marshal_success(void **state)
     sel.pcrSelections[1].pcrSelect[0] = 0xdd;
     sel.pcrSelections[1].pcrSelect[1] = 0xee;
 
-    ptr2 = (TPML_PCR_SELECTION *)(buffer);
-
     rc = Tss2_MU_TPML_PCR_SELECTION_Marshal(&sel, buffer, buffer_size, NULL);
-    assert_int_equal (rc, TSS2_RC_SUCCESS);
-    assert_int_equal (ptr2->count, HOST_TO_BE_32(2));
-    assert_int_equal (ptr2->pcrSelections[0].hash, HOST_TO_BE_16(TPM_ALG_SHA1));
-    assert_int_equal (ptr2->pcrSelections[0].sizeofSelect, 3);
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[0], 0xaa);
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[1], 0xbb);
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[2], 0xcc);
-    assert_int_equal (ptr2->pcrSelections[1].hash, HOST_TO_BE_16(TPM_ALG_SHA256));
-    assert_int_equal (ptr2->pcrSelections[1].sizeofSelect, 2);
-    assert_int_equal (ptr2->pcrSelections[1].pcrSelect[0], 0xdd);
-    assert_int_equal (ptr2->pcrSelections[1].pcrSelect[1], 0xee);
+
+    uint8_t expect [] = {
+        0, 0, 0, 2,
+        (uint8_t)(TPM_ALG_SHA1 / 256), (uint8_t)TPM_ALG_SHA1,
```

Comment:
This is quite confusing. (uint8_t)0x0004 / 256 will be 0 anyway. Why not just 0, 0x04?

Reply:
We wanted to make it explicit what is written into these... I guess it is a matter of style... We could replace it with
`0, 0x04, /* TPM_ALG_SHA1 */`
Would you prefer this ?

Reply:
It was just the div by 256. 0, 0x04, /* TPM_ALG_SHA1 */ looks good to me.